FeatureDetect/src/BuildOpenCV.cmake

```diff
@@ -0,0 +1,134 @@
+set(PROJECT_BUILD_DIR ${CMAKE_SOURCE_DIR}/build)
+
+set(OPENCV_VERSION 4.6.0)
+set(OPENCV_MODULES core,calib3d,imgproc,imgcodecs,features2d,highgui)
+set(OPENCV_DOWNLOAD_DIR ${PROJECT_BUILD_DIR}/opencv-download)
+set(OPENCV_BUILD_DIR ${PROJECT_BUILD_DIR}/opencv-build)
+
+# Append suffix to the build directory name
+if (BUILD_DIR_SUFFIX)
+    string(APPEND OPENCV_BUILD_DIR "-${BUILD_DIR_SUFFIX}")
+endif ()
+
+# Download and extract OpenCV sources
+if (NOT EXISTS ${OPENCV_DOWNLOAD_DIR}/opencv-${OPENCV_VERSION}/CMakeLists.txt)
+    message(STATUS "OpenCV not extracted into ${OPENCV_DOWNLOAD_DIR}")
+
+    if (NOT EXISTS ${OPENCV_DOWNLOAD_DIR}/${OPENCV_VERSION}.zip)
+        message(STATUS "OpenCV not downloaded into ${OPENCV_DOWNLOAD_DIR}")
+
+        message(STATUS "Downloading OpenCV")
+        file(
+                DOWNLOAD
+                https://github.com/opencv/opencv/archive/refs/tags/${OPENCV_VERSION}.zip
+                ${OPENCV_DOWNLOAD_DIR}/${OPENCV_VERSION}.zip
+                SHOW_PROGRESS
+        )
+        message(STATUS "Downloading OpenCV - done")
+    endif ()
+
+    message(STATUS "Extracting OpenCV")
+    file(
+            ARCHIVE_EXTRACT
+            INPUT ${OPENCV_DOWNLOAD_DIR}/${OPENCV_VERSION}.zip
+            DESTINATION ${OPENCV_DOWNLOAD_DIR}
+    )
+    message(STATUS "Extracting OpenCV - done")
+endif ()
+
+# Set toolchain-related flags
+if (CMAKE_GENERATOR)
+    list(APPEND TOOLCHAIN_ARGS -DCMAKE_GENERATOR=${CMAKE_GENERATOR})
+endif ()
+if (CMAKE_TOOLCHAIN_FILE)
+    list(APPEND TOOLCHAIN_ARGS -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE})
+endif ()
+if (CMAKE_C_COMPILER)
+    list(APPEND TOOLCHAIN_ARGS -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER})
+endif ()
+if (CMAKE_CXX_COMPILER)
+    list(APPEND TOOLCHAIN_ARGS -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER})
+endif ()
+
+# Configure OpenCV build
+set(
+        OPENCV_CMAKE_ARGS  # https://docs.opencv.org/4.5.5/db/d05/tutorial_config_reference.html
+        # General
+        -DBUILD_LIST=${OPENCV_MODULES}
+        # Bundled components
+        # -DWITH_GTK=ON
+        -DBUILD_TESTS=OFF
+        -DBUILD_PERF_TESTS=OFF
+        -DBUILD_ANDROID_PROJECTS=OFF
+        -DBUILD_ANDROID_EXAMPLES=OFF
+        -DBUILD_opencv_apps=OFF
+        -DBUILD_JAVA=OFF
+        -DBUILD_FAT_JAVA_LIB=OFF
+        -DBUILD_KOTLIN_EXTENSIONS=OFF
+        -DBUILD_OBJC=OFF
+        -DBUILD_opencv_python2=OFF
+        -DBUILD_opencv_python3=OFF
+        # Disabled dependencies
```

Comment:
I suggest to remove this for now and add a `TODO` to later disable the unwanted dependecies (when you'll know, which ones are unwanted)

Reply:
done, moved code to comments